### ISSUE TITLE
Bug inserting workers with decimal args

### ISSUE THREAD

**Issue by daniikpando** [sorentwo/oban#961]
### Environment

- **Oban PRO Version**: v1.1.3
- **Oban Version**: v2.15.4
- **PostgreSQL Version**: 13
- **Elixir & Erlang/OTP Versions**: 1.14.5

### Current Behavior

When I insert an Oban Worker with decimal arguments, It is inserting the argument like this:

```elixir
%{"coef" => 10000, "exp" => 0, "sign" => 1}
```

So, When the worker is ready to be processed, the Oban validation fails because It can't cast this `map()` to a `Decimal.t()` value.

Here's how I configured the Oban Worker:

```elixir
defmodule ObanWorker do
  use Oban.Pro.Worker, queue: :example
  
  args_schema do
    field :amount, :decimal, required: true
  end
  
  def process(%Oban.Job{args: %__MODULE__{amount: amount}}) do
    ...
    
    :ok
  end
end
```

Here's how Oban Worker returns the error.
```elixir
{:error, ":amount is invalid"}
```

### Expected Behavior

Before this issue, it was inserting the Decimal values as string in the DB, it is unclear to me why the behavior was affected because Ecto Changeset expects the value in this way.

**Comment by daniikpando** [comment]
Hi @sorentwo, any news about it?

**Comment by sorentwo** [comment]
@daniikpando This was fixed in Pro v1.1.4, released last week: https://oban.pro/releases/pro/v1.1#1.1.4

**Comment by daniikpando** [comment]
Ohhh, thanks @sorentwo!
